test/extensions/filters/network/common/fuzz/uber_per_readfilter.cc

```diff
@@ -157,6 +161,20 @@ void UberFilterFuzzer::checkInvalidInputForFuzzer(const std::string& filter_name
           "http_conn_manager trying to use Quiche which we won't fuzz here. Config:\n{}",
           config.DebugString()));
     }
+  } else if (filter_name == NetworkFilterNames::get().MongoProxy) {
+    envoy::extensions::filters::network::mongo_proxy::v3::MongoProxy& config =
+        dynamic_cast<envoy::extensions::filters::network::mongo_proxy::v3::MongoProxy&>(
+            *config_message);
+    if (config.has_delay() && config.mutable_delay()->has_header_delay()) {
+      // MongoProxy filter doesn't allow header_delay because it will pass nullptr to percentage()
```

Comment:
Is this field marked as unsupported in the API? Passing a `nullptr` to a function like this seems like a bug https://github.com/envoyproxy/envoy/blob/466245b50e6d475be841e1c8d5f254dd6ea41272/source/extensions/filters/common/fault/fault_config.cc#L15

Reply:
> Is this field marked as unsupported in the API? Passing a `nullptr` to a function like this seems like a bug
> 
> https://github.com/envoyproxy/envoy/blob/466245b50e6d475be841e1c8d5f254dd6ea41272/source/extensions/filters/common/fault/fault_config.cc#L15

I talked with author of this line @mattklein123, and he agrees that this is a config issue, and mongo's delay should be a fixed_delay.  For fixed_delay, nullptr is ok. But for header_delay, nullptr is not allowed. I think they extend the same base class and thus one allows nullptr and one doesn't.

see class FixedDelayProvider (source/extensions/filters/common/fault/fault_config.h) and class HeaderDelayProvider (source/extensions/filters/common/fault/fault_config.h)

Reply:
+1 to opting for the solution here https://github.com/envoyproxy/envoy/pull/12628/files
which guards against derefencing a nullptr passing into `percentage`.

Reply:
Other than this issue (which will be solved by committing  https://github.com/envoyproxy/envoy/pull/12628/), this change looks good to me.